Clinical-Genomics/cg
# Fix deletion without confirmation of other sample sheet existence

## Description
Right now a sample sheet is deleted from housekeeper if it fails validation when executing `cg demultiplex samplesheet create/create-all`. With this patch, the sample sheet will not be deleted until another sample sheet passes validation, either the one in the flow cell directory or a newly created sample sheet. If no other valid sample sheet can be fetched, the invalid sample sheet will remain in Housekeeper.

This patch avoids invalid sample sheets that have no samples in LIMS being deleted.


### Changed

- The deletion of the Housekeeper's sample sheet is moved after a validation of an existing sample sheet in two parts: In the creation of new sample sheets and in the verification of flow cell directory sample sheets.


### How to prepare for test

    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b fix-verify-sheet-before-deleting -a
    ```

### How to test

- [x] Do `cg demultiplex samplesheet create` when an invalid sample sheet is in housekeeper and there are no saples in LIMS

### Expected test outcome

- [x] Check that the sample sheet is not deleted despite being invalid

## Review

- [x] Tests executed by SD
- [x] "Merge and deploy" approved by IO
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a


## Implementation Plan
